examples/ex4.cpp

```diff
@@ -167,32 +186,41 @@ int main(int argc, char *argv[])
    }
    a->Assemble();
 
-   SparseMatrix A;
+   OperatorPtr A;
    Vector B, X;
    a->FormLinearSystem(ess_tdof_list, x, *b, A, X, B);
 
-   cout << "Size of linear system: " << A.Height() << endl;
+   cout << "Size of linear system: " << A->Height() << endl;
 
 #ifndef MFEM_USE_SUITESPARSE
-   // 10. Define a simple symmetric Gauss-Seidel preconditioner and use it to
-   //     solve the system A X = B with PCG.
-   GSSmoother M(A);
-   PCG(A, M, B, X, 1, 10000, 1e-20, 0.0);
+   // 11. Define a simple symmetric Gauss-Seidel preconditioner in the full
+   //     assembly case or Jacobi for partial assembly, and use it to solve
+   //     the system A X = B with PCG.
+   if (pa)
+   {
+      OperatorJacobiSmoother M(*a, ess_tdof_list);
+      PCG(*A, M, B, X, 1, 10000, 1e-20, 0.0);
+   }
+   else
+   {
+      GSSmoother M((SparseMatrix&)(*A));
+      PCG(*A, M, B, X, 1, 10000, 1e-20, 0.0);
+   }
 #else
-   // 10. If compiled with SuiteSparse support, use UMFPACK to solve the system.
+   // 11. If compiled with SuiteSparse support, use UMFPACK to solve the system.
    UMFPackSolver umf_solver;
```

Comment:
This checks should be changed like example 1, i.e if pa is enabled it shouldn't allow the direct solve and run by default the CG solver

Reply:
Thanks, I overlooked that case since I did not test with SUITESPARSE.